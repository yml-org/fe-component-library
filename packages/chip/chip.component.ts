import { ChipComponent } from './src/chip/chip.component';

declare global {
  interface HTMLElementTagNameMap {
    'ymlwebcl-chip': ChipComponent;
  }
}
