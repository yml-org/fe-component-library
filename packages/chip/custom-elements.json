{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "chip.component.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/index.ts",
      "declarations": [],
      "exports": [
        {
          "kind": "js",
          "name": "*",
          "declaration": {
            "name": "*",
            "package": "./chip/chip.component"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "dist/assets/index.js",
      "declarations": [
        {
          "kind": "variable",
          "name": "e"
        },
        {
          "kind": "variable",
          "name": "i"
        },
        {
          "kind": "variable",
          "name": "n"
        },
        {
          "kind": "variable",
          "name": "A",
          "default": "class extends Pt(null) {\n  constructor() {\n    super(...arguments), this.text = \"Label\", this.chipPartAttribute = \"webcl-chip\";\n  }\n  getContainerClassList() {\n    return \"[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]\";\n  }\n  render() {\n    return $t`\n      <div class=${this.getContainerClassList()} part=${this.chipPartAttribute}>\n        <slot name=${this.leftSlotName}></slot>\n        ${this.text ? this.text : this.text === \"\" ? \"\" : Rt(Ut`Label`)}\n        <slot name=${this.rightSlotName}></slot>\n      </div>\n    `;\n  }\n}"
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "i",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "n",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "js",
          "name": "ChipComponent",
          "declaration": {
            "name": "A",
            "module": "dist/assets/index.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/chip/chip.component.spec.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/chip/chip.component.ts",
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "ChipComponent",
          "members": [
            {
              "kind": "field",
              "name": "text",
              "type": {
                "text": "string | undefined"
              },
              "default": "'Label'"
            },
            {
              "kind": "field",
              "name": "rightSlotName",
              "type": {
                "text": "string | undefined"
              }
            },
            {
              "kind": "field",
              "name": "leftSlotName",
              "type": {
                "text": "string | undefined"
              }
            },
            {
              "kind": "field",
              "name": "chipPartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-chip'"
            },
            {
              "kind": "method",
              "name": "getContainerClassList",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "render",
              "privacy": "public"
            },
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]",
              "inheritedFrom": {
                "name": "TailwindElement",
                "module": "src/shared/tailwind.element.ts"
              }
            }
          ],
          "mixins": [
            {
              "name": "TailwindElement",
              "module": "/src/shared/tailwind.element"
            }
          ],
          "superclass": {
            "module": "src/chip/chip.component.ts"
          }
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ChipComponent",
          "declaration": {
            "name": "ChipComponent",
            "module": "src/chip/chip.component.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/constants/chip.variants.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "withSlot",
          "type": {
            "text": "ChipVariant"
          },
          "default": "{\r\n  rightSlotName: 'closeIcon',\r\n  leftSlotName: 'menuIcon',\r\n  chipPartAttribute : 'customizeChip'\r\n}"
        },
        {
          "kind": "variable",
          "name": "withTextProp",
          "type": {
            "text": "ChipVariant"
          },
          "default": "{\r\n  text : 'Chip Component',\r\n}"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "withSlot",
          "declaration": {
            "name": "withSlot",
            "module": "src/constants/chip.variants.ts"
          }
        },
        {
          "kind": "js",
          "name": "withTextProp",
          "declaration": {
            "name": "withTextProp",
            "module": "src/constants/chip.variants.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/shared/tailwind.element.ts",
      "declarations": [
        {
          "kind": "mixin",
          "description": "",
          "name": "TailwindElement",
          "members": [
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]"
            }
          ],
          "parameters": [
            {
              "name": "style",
              "type": {
                "text": "CSSStyleSheet | null"
              }
            }
          ]
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "TailwindElement",
          "declaration": {
            "name": "TailwindElement",
            "module": "src/shared/tailwind.element.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/types/chip.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/generated/locales/es-419.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "templates",
          "type": {
            "text": "object"
          },
          "default": "{\n      's5590dbf7e425789d': str`etiqueta`,\n    }"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "templates",
          "declaration": {
            "name": "templates",
            "module": "src/generated/locales/es-419.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/generated/locales/zh-Hans.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "templates",
          "type": {
            "text": "object"
          },
          "default": "{\n      's5590dbf7e425789d': str`标签`,\n    }"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "templates",
          "declaration": {
            "name": "templates",
            "module": "src/generated/locales/zh-Hans.ts"
          }
        }
      ]
    }
  ]
}
