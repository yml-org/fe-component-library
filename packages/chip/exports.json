{
  ".": "./src/index.js",
  "./src/*.js": "./src/*.js",
  "./package.json": "./package.json",
  "./chip.component.js": "./chip.component.js"
}
