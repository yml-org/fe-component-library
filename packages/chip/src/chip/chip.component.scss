.parent-container {
    chip-component::part(customizeChip) {
        background-color: grey;
        color: black;
        font-weight: 600;
        font-size: 20px;
        border: 2px solid black;
        border-radius: 24px;
    }
}
