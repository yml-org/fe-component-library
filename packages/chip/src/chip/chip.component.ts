import { html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { TailwindElement } from '../shared/tailwind.element';
import { msg, str } from '@lit/localize';

@customElement('ymlwebcl-chip')
export class ChipComponent extends TailwindElement(null) {
  @property({ type: String, reflect: true })
  text?: string = 'Label';

  @property({ type: String, reflect: true })
  rightSlotName?: string;

  @property({ type: String, reflect: true })
  leftSlotName?: string;

  @property()
  chipPartAttribute?: string = 'webcl-chip';

  private getContainerClassList() {
    return `[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]`;
  }

  public override render() {
    return html`
      <div class=${this.getContainerClassList()} part=${this.chipPartAttribute}>
        <slot name=${this.leftSlotName}></slot>
        ${this.text ? this.text : this.text === '' ? '' : msg(str`Label`)}
        <slot name=${this.rightSlotName}></slot>
      </div>
    `;
  }
}
