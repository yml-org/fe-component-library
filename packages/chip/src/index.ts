export * from './chip/chip.component';
