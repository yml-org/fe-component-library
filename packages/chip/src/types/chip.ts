export type ChipVariant = {
  text?: string;
  chipPartAttribute?: string;
  rightSlotName?: string;
  leftSlotName?: string;
};
