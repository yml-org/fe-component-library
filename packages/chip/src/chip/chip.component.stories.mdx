import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './chip.component.ts';
import './chip.component.scss';
import { withTextProp, withSlot } from '../constants/chip.variants';

<Meta title="Components/Chip" />

# Chip Component

The Chip component is a Lit component that renders a chip component that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name              | Type   | Description                                                                                   |
| ---------------------- | ------ | --------------------------------------------------------------------------------------------- |
| text                   | string | text within the chip component                                                                |
| chipPartAttribute      | string | The part name for chip component                                                              |
| rightSlotName          | string | Name given to slot that renders an HTMLElement on the right of the text within chip component |
| leftSlotName           | string | Name given to slot that renders an HTMLElement on the left of the text within chip component  |

# Examples

Default Chip Component

export const defaultChipTemplate = () =>
  html`<chip-component></chip-component>`;

<Canvas>
  <Story name="default chip">{defaultChipTemplate.bind({})}</Story>
</Canvas>

Chip Component with text prop

export const textTemplate = ({ text }) =>
  html`<chip-component .text=${text}></chip-component>`;

<Canvas>
  <Story name="text prop" args={withTextProp}>
    {textTemplate.bind({})}
  </Story>
</Canvas>


Chip Component with right slot element

export const chipRightSlotTemplate = ({ rightSlotName }) =>
  html`<chip-component
    .rightSlotName=${rightSlotName}
  >
   <div slot=${rightSlotName}>
      <svg
        width="20px"
        height="20px"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 22C17.5 22 22 17.5 22 12C22 6.5 17.5 2 12 2C6.5 2 2 6.5 2 12C2 17.5 6.5 22 12 22Z"
          stroke="#292D32"
          stroke-width="1.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <g opacity="0.4">
          <path
            d="M9.16992 14.8299L14.8299 9.16992"
            stroke="#292D32"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M14.8299 14.8299L9.16992 9.16992"
            stroke="#292D32"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </g>
      </svg>
    </div>
  </chip-component>`;

  <Canvas>
  <Story name="chip with right slot" args={withSlot}>
    {chipRightSlotTemplate.bind({})}
  </Story>
</Canvas>

Chip Component with left slot element

export const chipLeftSlotTemplate = ({ leftSlotName }) =>
  html`<chip-component
    .leftSlotName=${leftSlotName}
  >
    <div slot=${leftSlotName}>
      <svg
        width="20px"
        height="20px"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 17H20M4 12H20M4 7H20"
          stroke="#000000"
          stroke-width="1.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </chip-component>`;

  <Canvas>
  <Story name="chip with left slot" args={withSlot}>
    {chipLeftSlotTemplate.bind({})}
  </Story>
</Canvas>

Chip Component with slot elements

export const chipSlotTemplate = ({ rightSlotName, leftSlotName }) =>
  html`<chip-component
    .rightSlotName=${rightSlotName}
    .leftSlotName=${leftSlotName}
  >
    <div slot=${rightSlotName}>
      <svg
        width="20px"
        height="20px"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 22C17.5 22 22 17.5 22 12C22 6.5 17.5 2 12 2C6.5 2 2 6.5 2 12C2 17.5 6.5 22 12 22Z"
          stroke="#292D32"
          stroke-width="1.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <g opacity="0.4">
          <path
            d="M9.16992 14.8299L14.8299 9.16992"
            stroke="#292D32"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M14.8299 14.8299L9.16992 9.16992"
            stroke="#292D32"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </g>
      </svg>
    </div>
    <div slot=${leftSlotName}>
      <svg
        width="20px"
        height="20px"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 17H20M4 12H20M4 7H20"
          stroke="#000000"
          stroke-width="1.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </chip-component>`;

<Canvas>
  <Story name="chip with slot" args={withSlot}>
    {chipSlotTemplate.bind({})}
  </Story>
</Canvas>

Customized Chip Component

export const withCustomStylingTemplate = ({ chipPartAttribute }) =>
  html` <div class="parent-container">
    <chip-component .chipPartAttribute=${chipPartAttribute}> </chip-component>
  </div>`;

<Canvas>
  <Story name="with custom styling" args={withSlot}>
    {withCustomStylingTemplate.bind({})}
  </Story>
</Canvas>

Customized Chip Component with Customized slot element

export const withSlotCustomStylingTemplate = ({
  rightSlotName,
  leftSlotName,
  chipPartAttribute,
}) =>
  html` <div class="parent-container">
    <chip-component
      .rightSlotName=${rightSlotName}
      .leftSlotName=${leftSlotName}
      .chipPartAttribute=${chipPartAttribute}
    >
      <div style="margin-left:25px" slot=${rightSlotName}>
        <svg
          width="20px"
          height="20px"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 22C17.5 22 22 17.5 22 12C22 6.5 17.5 2 12 2C6.5 2 2 6.5 2 12C2 17.5 6.5 22 12 22Z"
            stroke="#292D32"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g opacity="0.4">
            <path
              d="M9.16992 14.8299L14.8299 9.16992"
              stroke="#292D32"
              stroke-width="1.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M14.8299 14.8299L9.16992 9.16992"
              stroke="#292D32"
              stroke-width="1.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </g>
        </svg>
      </div>
      <div  style="margin-right:25px" slot=${leftSlotName}>
        <svg
          width="20px"
          height="20px"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 17H20M4 12H20M4 7H20"
            stroke="#000000"
            stroke-width="1.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </chip-component>
  </div>`;

<Canvas>
  <Story name="custom styling with slot" args={withSlot}>
    {withSlotCustomStylingTemplate.bind({})}
  </Story>
</Canvas>
