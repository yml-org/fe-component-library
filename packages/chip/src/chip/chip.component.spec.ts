import { LitElement } from 'lit-element';
import { withTextProp, withSlot } from '../constants/chip.variants';

describe('chip-component', () => {
  const CHIP_COMPONENT = 'chip-component';
  let chipElement: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    chipElement = window.document.createElement(CHIP_COMPONENT) as LitElement;
    document.body.appendChild(chipElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(CHIP_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(CHIP_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('sets the text property correctly', async () => {
    chipElement['text'] = withTextProp.text;
    await chipElement.updateComplete;
    const elem = getShadowRoot(CHIP_COMPONENT).querySelector('div').innerText;
    expect(elem).toBe(withTextProp.text);
  });

  it('sets the chipPartAttribute property correctly', async () => {
    chipElement['chipPartAttribute'] = withSlot.chipPartAttribute;
    await chipElement.updateComplete;
    const elem = getShadowRoot(CHIP_COMPONENT).querySelector('div');
    expect(elem.getAttribute('part')).toBe(withSlot.chipPartAttribute);
  });

  it('renders the slot elements correctly', async () => {
    chipElement['chipPartAttribute'] = withSlot.rightSlotName;
    getShadowRoot(CHIP_COMPONENT)
      .querySelector('div')
      .querySelector('div')
      .querySelector('slot').innerHTML =
      '<h2 class="slot-test">The chip Component</h2>';
    await chipElement.updateComplete;
    const elem = getShadowRoot(CHIP_COMPONENT)
      .querySelector('div')
      .querySelector('div')
      .querySelector('slot')
      .querySelector('h2');
    expect(elem.classList.contains('slot-test')).toBe(true);
  });
});
