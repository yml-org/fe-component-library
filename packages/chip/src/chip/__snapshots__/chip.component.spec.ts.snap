// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chip-component matches component snapshot 1`] = `
v {
  "_$AL": Map {},
  "_$Do": U {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": mt {
      "_$AD": N {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 1,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 2,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 3,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 4,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        R {
          "_$AH": "[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]"
            part="webcl-chip"
          >
            
        
            <div
              class="mr-[6px]"
              part="webcl-chip-leftSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>
            
        
            <!--?lit$97514775$-->
            Label
            
        
            <div
              class="ml-[6px]"
              part="webcl-chip-rightSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>
            
      
          </div>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "webcl-chip",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]"
            part="webcl-chip"
          >
            
        
            <div
              class="mr-[6px]"
              part="webcl-chip-leftSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>
            
        
            <!--?lit$97514775$-->
            Label
            
        
            <div
              class="ml-[6px]"
              part="webcl-chip-rightSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>
            
      
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "webcl-chip-leftSlot",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="mr-[6px]"
            part="webcl-chip-leftSlot"
          >
            
          
            <slot
              name=""
            />
            
        
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        U {
          "_$AA": <!--?lit$97514775$-->,
          "_$AB": 
        ,
          "_$AH": "Label",
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        R {
          "_$AH": "webcl-chip-rightSlot",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="ml-[6px]"
            part="webcl-chip-rightSlot"
          >
            
          
            <slot
              name=""
            />
            
        
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__chipPartAttribute": "webcl-chip",
  "__leftSlotPartAttribute": "webcl-chip-leftSlot",
  "__rightSlotPartAttribute": "webcl-chip-rightSlot",
  "__text": "Label",
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": U {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": mt {
        "_$AD": N {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 1,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 2,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 3,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 4,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          R {
            "_$AH": "[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]"
              part="webcl-chip"
            >
              
        
              <div
                class="mr-[6px]"
                part="webcl-chip-leftSlot"
              >
                
          
                <slot
                  name=""
                />
                
        
              </div>
              
        
              <!--?lit$97514775$-->
              Label
              
        
              <div
                class="ml-[6px]"
                part="webcl-chip-rightSlot"
              >
                
          
                <slot
                  name=""
                />
                
        
              </div>
              
      
            </div>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "webcl-chip",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="[word-wrap: break-word] inline-flex flex-row items-center my-[5px] pt-[4px] pb-[6px] px-[12px]"
              part="webcl-chip"
            >
              
        
              <div
                class="mr-[6px]"
                part="webcl-chip-leftSlot"
              >
                
          
                <slot
                  name=""
                />
                
        
              </div>
              
        
              <!--?lit$97514775$-->
              Label
              
        
              <div
                class="ml-[6px]"
                part="webcl-chip-rightSlot"
              >
                
          
                <slot
                  name=""
                />
                
        
              </div>
              
      
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "webcl-chip-leftSlot",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="mr-[6px]"
              part="webcl-chip-leftSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          U {
            "_$AA": <!--?lit$97514775$-->,
            "_$AB": 
        ,
            "_$AH": "Label",
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          R {
            "_$AH": "webcl-chip-rightSlot",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="ml-[6px]"
              part="webcl-chip-rightSlot"
            >
              
          
              <slot
                name=""
              />
              
        
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
